src/core/BoardStateSupervisor.ts

```diff
@@ -0,0 +1,89 @@
+import { BoardState, Color, PieceType } from "constants/enums";
+import Board from "./Board";
+import Move from "./Move";
+import MoveGenerator from "./MoveGenerator";
+import King from "./pieces/King";
+import Piece from "./pieces/Piece";
+
+class BoardStateSupervisor {
```

Comment:
What do you think about renaming it do BoardStateManager or even GameStateManager? In fact board is just place where we put pieces, and check or checkmate are general states of the game that also includes information about the player that is checked etc

Reply:
Done, also changed BoardState to GameState.